src/lib/Makefile.am

```diff
@@ -3,8 +3,18 @@
 SUBDIRS = plugin_apis
 
 lib_LTLIBRARIES = libblockdev.la
-libblockdev_la_CFLAGS = $(GLIB_CFLAGS)
-libblockdev_la_LIBADD = $(GLIB_LIBS) ${builddir}/../utils/libbd_utils.la -ldl
+libblockdev_la_CFLAGS =
+libblockdev_la_LIBADD =
+if WITH_BTRFS
+libblockdev_la_CFLAGS += $(GOBJECT_CFLAGS)
+libblockdev_la_LIBADD += $(GOBJECT_LIBS)
+endif
+if WITH_LVM
+libblockdev_la_CFLAGS += $(GOBJECT_CFLAGS)
+libblockdev_la_LIBADD += $(GOBJECT_LIBS)
+endif
+libblockdev_la_CFLAGS += $(GLIB_CFLAGS)
+libblockdev_la_LIBADD += $(GLIB_LIBS) ${builddir}/../utils/libbd_utils.la -ldl
```

Comment:
This was just a quick hack. The if statements also likely should be nested to avoid adding GOBJECT_LIBS/CFLAGS twice if both BTRFS and LVM is enabled. I thought it ended up ugly either way, so didn't bother trying to do a clean version.... you decide which style you think is the least ugly.